test/helpers.cpp

```diff
@@ -179,5 +162,7 @@ void PreprocessorHelper::preprocess(Preprocessor &preprocessor, const char code[
     // Tokenizer..
     tokenizer.list.createTokens(std::move(tokens2));
 
-    preprocessor.setDirectives(tokens1);
+    const Preprocessor preprocessor(tokenizer.getSettings());
+    std::list<Directive> directives = preprocessor.createDirectives(tokens1);
+    tokenizer.setDirectives(std::move(directives));
```

Comment:
or:

    tokenizer.setDirectives(preprocessor.createDirectives(tokens1));

Reply:
Yeah - I wanted to keep it separate for readability.

Reply:
well then I personally would have used a const reference instead. It's not necessary with a std::move here afaik.

Reply:
That would have introduced an unnecessary copy.

Reply:
const std::list<Directive>& directives = preprocessor.createDirectives(tokens1);
    tokenizer.setDirectives(directives);

there is a copy with your code also as far as I see.